### ISSUE TITLE
Not Possible to Transmit on Multiple Radios

### ISSUE THREAD

**Issue by Eddie686** [ciribob/DCS-SimpleRadioStandalone#158]
At present it is only possible to transmit on a single radio, this presents an unrealistic limitation for AWACS/ATC roles where transmission on multiple frequencies is common place (e.g VHF and UHF channels for ATC).

In addition some aircraft can transmit on multiple radios simultaneously (those that can't only have this limitation due to PTT configuration for the most part rather than a radio set limitation)

Add ability to transmit on multiple radios.

**Comment by 132nd-Entropy** [comment]
Hello, we just tested on NTTR with SRS 1.2.9.4

Issue persists, as soon as someone talks on a second radio, comms get totally garbled. Hansolo made a recording which he will upload.

**Comment by ciribob** [comment]
Transmitting on multiple radios simultaneously requires a complete rewrite of how audio is encoded, sent and distributed but it's definitely something I'd like to do but its not going to be for a while. Sorry!

@132nd-Entropy do you mean this bug? #137 If you could add the audio there that would be very helpful!

**Comment by 132nd-Entropy** [comment]
@ciribob oops sorry, I didnt read the original post properly. Of course my comment is regarding bug 148 not this one. sorry :)

**Comment by ciribob** [comment]
No worries :)

**Comment by gyrovague** [comment]
FWIW, F-14 also has the ability to simultaneously transmit on two radios, so this feature would be handy there too.

I guess to support this, SRS will need to change to have PTT in the exports be a per-radio thing (so that multiple can have PTT true at once), and the outgoing audio packets need to have a list of dest frequencies instead of just one, and the server needs to iterate that list in each incoming packet, and find any matching listeners to distribute the packet to etc.. or I guess the client could send an audio packet multiple times, once with each different dest frequency, but this would be a bit less efficient (maybe not a huge problem though (VoIP bandwidth usage is quite low with the codecs SRS uses), and would require less overall SRS changes probably).

**Comment by ciribob** [comment]
First part of this is done -AWACS can transmit on multiple radios simultaneously

**Comment by ciribob** [comment]
@gyrovague can you give some details on how you do the simultaneous transmission on the F14 - SRS supports this as of last july 2018 but I forgot to follow up for any specific F14 details

**Comment by gyrovague** [comment]
Hi @ciribob , apologies for only noticing your message now. The RIO in the F-14 can select which radio transmits when PTT is pressed, and one of the selections he can make is "both"

I _think_ the exports we send to SRS already indicate that individually for each radio, I'll double check.  Might be that only the SRS lua function for F-14 needs to change a bit?

**Comment by gyrovague** [comment]
I think you might be able to just change this logic in the main SRS lua file:
```
    if (ARC182_ptt) then
        _data.selected = 2 -- radios[3] ARC-182
        _data.ptt = true
    elseif (ARC159_ptt) then
        _data.selected = 1 -- radios[2] ARC-159
        _data.ptt = true
    elseif (intercom_transmit) then
```

those ptt can be true independently of each, so if RIO selects "both" in the cockpit, both of them will be true.

**Comment by gyrovague** [comment]
I'm not sure how SRS now handles the export w.r.t. multiple simultaneous PTT though, see my comment above: https://github.com/ciribob/DCS-SimpleRadioStandalone/issues/158#issuecomment-382349038

**Comment by gyrovague** [comment]
You maybe need to add something like a `selected_bitmap` member in that update object, and default it to 0 for backward compatibility with all other radios (i.e. if it's 0, use the `selected` member or something like that), and for F-14 make it set the bits corresponding to the radios, e.g.
```
    if (ARC182_ptt) then
        _data.selected_bitmap = _data.selected_bitmap +  4 -- radios[3] ARC-182
        _data.ptt = true
    elseif (ARC159_ptt) then
        _data.selected_bitmap = _data.selected_bitmap +  2 -- radios[2] ARC-159
        _data.ptt = true
    elseif (intercom_transmit) then
        _data.selected_bitmap = _data.selected_bitmap +  1 -- radios[1] intercom
        _data.ptt = true
    else
```
(using `+` here with values like 1,2,4,8 since lua doesn't have bitwise operators I think)

and then have the C# code recognise that to figure out why radios in the array of radios to transmit on.
*handwave*

**Comment by ciribob** [comment]
Ah gotcha! 

Thank you very much. Sorry, should've checked the cockpit as I thought it was more than just a switch 😀

There is a flag on each radio called simul that I think I can just send so the ptt will apply to all radios with that flag. 

Should be able to get in for the next release.

**Comment by ciribob** [comment]
Are there any other aircraft that support this?

**Comment by ciribob** [comment]
@gyrovague will be in the next SRS release - pull request here: https://github.com/ciribob/DCS-SimpleRadioStandalone/pull/394

**Comment by gyrovague** [comment]
cool, I'm adding a comment to the pull request now.

**Comment by ciribob** [comment]
Will close this now as the pull request will go out this week most likely
Implemented in #394
